sdk/go/common/resource/config/map_test.go

```diff
@@ -1253,7 +1253,6 @@ func TestSetFail(t *testing.T) {
 		{Key: "my:root["},
 		{Key: `my:root["nested]`},
 		{Key: "my:root.array[abc]"},
-		{Key: "my:root.[1]"},
```

Comment:
This was what we were trying to prevent with my previous PR on this topic.  But I guess this should then only fail at the time when we parse it from the config file/command line, but not if we construct property paths internally?  Mostly just curious as this was a test that existed for a while before we removed and re-added it.

Reply:
I'm thinking I'll just allow this for now. Just a typo we'll have to accept.

Reply:
:+1: yeah I don't think it's the end of the world, I was mostly just curious and trying to spell out my understanding of what's going on here.  Accepting this typo seems not too bad to me.